subsys/bluetooth/controller/ll_sw/ull_llcp_local.c

```diff
@@ -382,13 +382,19 @@ static void lr_st_disconnect(struct ll_conn *conn, uint8_t evt, void *param)
 static void lr_st_idle(struct ll_conn *conn, uint8_t evt, void *param)
 {
 	struct proc_ctx *ctx;
+	enum llcp_proc curr_proc;
 
 	switch (evt) {
 	case LR_EVT_RUN:
 		ctx = llcp_lr_peek(conn);
 		if (ctx) {
+			/*
+			 * since the call to lr_act_run may release the context we need to remember
+			 * which procedure we are running
+			 */
+			curr_proc = ctx->proc;
 			lr_act_run(conn);
-			if (ctx->proc != PROC_TERMINATE) {
+			if (curr_proc != PROC_TERMINATE) {
 				lr_set_state(conn, LR_STATE_ACTIVE);
```

Comment:
Would `lr_set_state` try to access any released context? Why I ask, `lr_act_run(conn)` has similar parameter signature of passing `conn` and accessing the `ctx` to release it.

Reply:
I checked, lr_set_state does not access the ctx